src/rp2_common/pico_rand/include/pico/rand.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef _PICO_RAND_H
+#define _PICO_RAND_H
+
+#include "pico.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** \file pico/rand.h
+ *  \defgroup pico_rand pico_rand
+ *
+ * Random Number Generator API
+ *
+ * This module generates random numbers at runtime via a couple of entropy
+ * sources and uses those sources to modify the state of a 128-bit 'Pseudo
+ * Random Number Generator' implemented in software.
+ *
+ * The random numbers to be supplied are read from the PRNG which is used
+ * to help provide a large number space.
+ *
+ * The main source of entropy at run time is a provided by ROSC 'random bit'
+ * samples but system time is also used to add a little extra entropy.
+ *
+ * All entropy sources are hashed before application to the PRNG state machine.
+ *
+ * Note: The *first* time a random number is requested, the 128-bit PRNG state
+ * must be seeded.  The seed generation takes approximately 1 millisecond with
+ * subsequent random numbers taking between 10 and 20 microseconds to generate.
```

Comment:
Presumably this time is also dependent upon what `PICO_RAND_MIN_ROSC_SAMPLE_TIME_US` has been set to?

Reply:
Absolutely